### PR TITLE
Naive scheduling

### LINKED ISSUE #12: Implement naive scheduling

### PR DESCRIPTION
**Pull request by Mikkel-Boje**
resolves #12.
Made an implementation for the naïve scheduling algorithm, also created the needed data types which the algorithm is dependent upon.

### CODE REVIEW

File: backend/src/data_model/graph.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+use chrono::Duration;
+use protocol::time::DateTimeUtc;
+pub struct DescreteGraph {
+    values: Vec<f64>,
+    time_delta: Duration,
+    start_time: DateTimeUtc,
+}
+
+impl DescreteGraph {
```
[the remark this answers is not in the document]
Comment (Mikkel-Boje) [reply backend/src/data_model/graph.rs:9]: used in tests, in scheduler file. Dunno why its saying it isnt

File: backend/src/scheduling/scheduler.rs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+fn adjust_graph_for_time_delta(timeslots: usize, graph_values: &[f64]) -> Vec<f64> {
+    graph_values
+        .windows(timeslots)
+        .map(|window| window.iter().sum())
+        .collect()
+}
+
+fn checked_duration_to_i64(dur: Duration) -> i64 {
+    i64::from(Milliseconds::from(dur))
+}
```
Comment (mads256h) [inline backend/src/scheduling/scheduler.rs:34]:
```suggestion
fn checked_duration_to_i64(duration: Duration) -> i64 {
    i64::from(Milliseconds::from(duration))
}
```

File: backend/src/scheduling/scheduler.rs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        let events = scheduler.schedule(graph, tasks).unwrap();
+
+        assert_eq!(
+            events[0],
+            UnPublishedEvent {
+                task_id: 1.into(),
+                start_time: start + Duration::seconds(1),
+            }
+        )
+    }
```
Comment (mads256h) [inline backend/src/scheduling/scheduler.rs:115]: Kill

**Review by mads256h — APPROVED** [review]

File: backend/src/scheduling/scheduler.rs (hunk 2 above)
Comment (Mikkel-Boje) [reply backend/src/scheduling/scheduler.rs]: Changed as recommended

File: backend/src/scheduling/scheduler.rs (hunk 3 above)
Comment (Mikkel-Boje) [reply backend/src/scheduling/scheduler.rs]: Changed as recommended

**Review by Darkros1245 — APPROVED** [review]

**Pull request merged**